src/pyramid/session.py

```diff
@@ -44,10 +44,24 @@ def changed(session, *arg, **kw):
 
 
 class PickleSerializer(object):
-    """ A serializer that uses the pickle protocol to dump Python
-    data to bytes.
+    """
+    .. deprecated:: 2.0
+
+    .. warning::
+
+    In :app:`Pyramid` 2.0 the default ``serializer`` option changed to
+    use :class:`pyramid.session.JSONSerializer`, and ``PickleSerializer`
+    has been been removed from active Pyramid code.
+
+    Pyramid will require JSON-serializable objects in :app:`Pyramid` 2.0.
+
+    Please see :ref:`pickle_session_deprecation`.
+
+    Also, please see: #2709, #3353, #3413
+
+    A serializer that uses the pickle protocol to dump Python data to bytes.
```

Comment:
This block needs to be indented 4 spaces to fall under the `.. warning::` reStructuredText directive.

Also I do not think that we have enabled the Sphinx extension that will convert issue or PR numbers to hyperlinks, so please use the actual links to those issues/PRs.

Reply:
ok; I'm going to migrate that info into the session docs too.

Reply:
I think it is also good practice to put links to the issue/PRs in the change notes instead of docstrings. I would look at change notes first when I see deprecation warnings, and I can refer to issue tracker for further information.

Reply:
Noted for next time!